test/api.test.js

```diff
@@ -535,6 +545,74 @@ describe("api.js", () => {
             }]);
         });
 
+        it("should fail by timeout during rendition processing", async () => {
+            delete process.env.DISABLE_IO_EVENTS_ON_TIMEOUT;
+
+            const receivedMetrics = MetricsTestHelper.mockNewRelic();
+            testUtil.nockIOEvent({
+                type: "rendition_failed",
+                errorReason: "GenericError",
+                errorMessage: 'Timeout',
+                rendition: {
+                    fmt: "png"
+                },
+                source: "https://example.com/MySourceFile.jpg"
+            });
+
+            const main = worker(async function(source, rendition) {
+                fs.writeFileSync(rendition.path, testUtil.RENDITION_CONTENT);
+                await sleep(500);
+                console.log('waiting...');
+                return Promise.resolve();
+            });
+            assert.equal(typeof main, "function");
+            process.env.__OW_DEADLINE = Date.now() + 300;
```

Comment:
You can probably set this down to 100 ms or even 50 ms to make the test faster with the same result.